### DISCUSSION TITLE
Get current user name.

### DISCUSSION THREAD

**Discussion by adriankarlen** [wezterm/wezterm#5514] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

How can I get the current username for the session via API? I know I can use `wezterm cli list-clients` and that includes the name, but I want to use it in my config, so I'd like to use it from the API.

Atm I am executing a whoami as a `io.popen("whoami"):read("*a"):gsub("\n", "")` but on windows this leads to a new cli-window popping up while executing the command.

Thanks :)

**Comment by wez** [comment, marked as answer]
I would probably just read it from the environment:

```lua
local username = os.getenv('USER') or os.getenv('LOGNAME') or os.getenv('USERNAME')
```

POSIX specifies that `LOGNAME` holds the username, although most unix systems also populate `USER`.
Windows sets `USERNAME` to the username, so the above tries each in turn.
